Eevee.Sleep.Bot/Controllers/Mongo/UserRoleController.cs

```diff
@@ -0,0 +1,58 @@
+
+
+
+using Eevee.Sleep.Bot.Models;
+using MongoDB.Driver;
+
+namespace Eevee.Sleep.Bot.Controllers.Mongo;
+
+public static class UserRoleContoller
+{
+    public static async Task SaveRoles(string userId, ulong[] roles)
+    {
+        var collection = MongoConst.RoleCollection;
+        var filter = Builders<UserRoleModel>.Filter.Eq("userId", userId);
+        var update = Builders<UserRoleModel>.Update.Set("roles", roles);
+        await collection.UpdateOneAsync(filter, update, new UpdateOptions { IsUpsert = true });
+    }
+
+    public static async Task RemoveRoles(string userId, ulong[] roles)
+    {
+        var collection = MongoConst.RoleCollection;
+        var filter = Builders<UserRoleModel>.Filter.Eq("userId", userId);
+        foreach (var role in roles)
+        {
+            var update = Builders<UserRoleModel>.Update.Pull("roles", role);
+
+            await collection.UpdateOneAsync(filter, update);
+        }
+    }
+
+    public static async Task AddRoles(string userId, ulong[] roles)
+    {
+        foreach (var role in roles)
+        {
+            await AddRole(userId, role);
+        }
+    }
+
+    private static async Task AddRole(string userId, ulong role)
+    {
+        var collection = MongoConst.RoleCollection;
+        var filter = Builders<UserRoleModel>.Filter.And(
+            Builders<UserRoleModel>.Filter.Eq("userId", userId),
+            Builders<UserRoleModel>.Filter.Ne("roles", role)
+        );
+        var update = Builders<UserRoleModel>.Update.Push("roles", role);
```

Comment:
Use [$addToSet](https://www.mongodb.com/docs/manual/reference/operator/update/addToSet/) to avoid pushing duplicated roles, although unlikely to happen

Reply:
Fixed!
[50ebdf0](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/50ebdf09885e8a7607f0ff594bd55ae3c9e0799d)